### ISSUE TITLE
[Feature Request] Loading external libraries

### ISSUE THREAD

**Issue by jnbooth** [khvzak/mlua#61]
I've been using mlua to provide scripting for my application, and it's doing a fantastic job. However, I would like users to have access to a few other external libraries, such as [lbc](https://github.com/LuaDist/lbc) and [LuaSQLite3](http://lua.sqlite.org/index.cgi/doc/tip/doc/lsqlite3.wiki). At present, the only way to do that seems to be initializing the state myself, calling the various initialization functions on it, and then passing that state to `Lua::init_from_ptr`. However, that seems like not a particularly good approach for several reasons:

- I can't set `.ephemeral = false`
- I'm using LuaJIT, so I don't have access to `luaL_requiref` (I would use your Rust port of compat-5.3 if it were a standalone package!)
- Loading the standard libraries is kind of a drag, and it seems like they should be loaded _after_ `init_from_ptr`

If there were some kind of `unsafe fn new_with_libraries` that accepts arguments for `luaL_requiref`, that would solve the problem. I suppose a more brute-force approach would be to simply accept a closure of `FnOnce(&mut lua_State)` during initialization. I of course understand that loading arbitrary external libraries would break all safety guarantees, but I think it would still be preferable to using `init_from_ptr`, which as far as I can tell is my only option right now.

**Comment by khvzak** [comment]
Am I correct that you want to statically link libraries such as LuaSQlite3 to your application?

**Comment by jnbooth** [comment]
I would probably include some (or all, if necessary) of the libraries directly as source code, because some of them (e.g. lbc) are only available as such.

**Comment by khvzak** [comment]
I agree that this functionality could be useful and working on creating more generic implementation.

My idea is to provide a method to create `Function` from a raw C  function (`ffi::lua_CFunction` type) and then it should be easy to add any required functionality.
For example, you could set `package.preload[modname]` to a C function and then Lua `require` would magically work.

**Comment by khvzak** [comment]
I pushed a commit d3024d0a93c989f8e101907e3ed5c26c146468cc to the dev branch with new functions: `load_from_function` and `create_c_function`. Could you check please that this provides a solution for your case?

Usage example:
```
let lsqlite3_open = unsafe { lua.create_c_function(luaopen_lsqlite3) }?;
let _: Value = lua.load_from_function("lsqlite3", lsqlite3_open)?;
```

**Comment by jnbooth** [comment]
Yep, that does it. Thank you so much!

**Comment by khvzak** [comment]
This is published now in v0.6.2
